### PR TITLE
STORM-188. Allow user to specifiy full configuration path when running storm command

### PR DESCRIPTION
**Pull request by harshach**

### CODE REVIEW

**Comment by harshach** [comment]
This PR is taken from https://github.com/apache/storm/pull/120#issuecomment-69435220

@clockfly @revans2 can you please take a look.

**Comment by clockfly** [comment]
+1

File: storm-core/src/jvm/backtype/storm/utils/Utils.java [hunk 1]
```diff
@@ -135,35 +137,68 @@ public static void sleep(long millis) {
...
+            throw new IOException(
+                    "Could not find config file, name not specified");
+        }
+
+        HashSet<URL> resources = new HashSet<URL>(findResources(configFilePath));
+        if (resources.isEmpty()) {
+            File configFile = new File(configFilePath);
+            if (configFile.exists()) {
+                return new FileInputStream(configFile);
+            }
```
Comment (lazyval) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: Why not throw IOException, when path contains no actual file?

Comment (clockfly) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: @lazyval 

mustExist flag will control whether we should fail the method.
···public static Map findAndReadConfigFile(String name, boolean mustExist) ···

When we cannot find a resource, we will return null; then we check whether mustExist flag is enabled to determine whether we should throw RuntimeException.

File: storm-core/src/jvm/backtype/storm/utils/Utils.java [hunk 2]
```diff
@@ -135,35 +137,68 @@ public static void sleep(long millis) {
     }
 
     public static Map findAndReadConfigFile(String name, boolean mustExist) {
+        InputStream in = null;
+        Boolean confFileEmpty = false;
```
Comment (lazyval) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: no need in boxed value

File: storm-core/src/jvm/backtype/storm/utils/Utils.java (hunk 1 above)
Comment (lazyval) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: Yeah, I just a bit concerned that flow is driven by both exceptions and flag/null.

File: storm-core/src/jvm/backtype/storm/utils/Utils.java (hunk 2 above)
Comment (harshach) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: @lazyval  what you mean by boxed value?

Comment (lazyval) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: There are two variables types in Java: primitive (like boolean, int, long, short and so on), and their boxed counterparts (Boolean, Integer, Long, ...), which are seen as reference types. It's minor, but unless you're working with collections, there is no need to use later one.

Comment (harshach) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: @lazyval  I know about primitives and objects in java . Yes makes sense to use a boolean.

**Comment by harshach** [comment]
@clockfly @lazyval updated the patch for fixing the last comment.

File: storm-core/src/jvm/backtype/storm/utils/Utils.java (hunk 1 above)
Comment (harshach) [inline storm-core/src/jvm/backtype/storm/utils/Utils.java]: @lazyval there are cases where storm wants to use defaults.yaml if there is not storm.yaml. If one wants to do a quick test they can just run the storm cluster by not having any storm.yaml . Hence the reason we have mustexist as a flag to indicate if we want to throw an error or not.

**Comment by lazyval** [comment]
:+1:

**Comment by revans2** [comment]
I think overall it is good and I am +1.

**Pull request merged**